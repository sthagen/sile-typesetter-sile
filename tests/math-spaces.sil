\begin[class=plain]{document}
\use[module=packages.retrograde,target=v0.15.0]

\use[module=packages.math]

Positive space commands:

\begin{math}
  \mi{thin}\thinspace
  \mi{med}\medspace
  \mi{thick}\thickspace
  \mi{en}\enspace
  \mi{quad}\quad
  \mi{qquad}\qquad
  \mi{end}
\end{math}

\par
\medskip
Positive space commands (aliases):

\begin{math}
  \mi{thin}\,
  \mi{med}\>
  \mi{thick}\;
  \mi{en}\enskip
  \mi{end}
\end{math}

\par
\medskip
Positive spaces in subscripts:

\begin{math}
  x_{
    \mi{thin}\thinspace
    \mi{med}\medspace
    \mi{thick}\thickspace
    \mi{en}\enspace
    \mi{quad}\quad
    \mi{qquad}\qquad
    \mi{end}_{
      \mi{thin}\thinspace
      \mi{med}\medspace
      \mi{thick}\thickspace
      \mi{en}\enspace
      \mi{quad}\quad
      \mi{qquad}\qquad
      \mi{end}
    }
  }
\end{math}

\medskip
Negative space commands:

\begin{math}
  \mi{nthin}\negthinspace
  \mi{nmed}\negmedspace
  \mi{nthick}\negthickspace
  \mi{end}
\end{math}

\medskip
Negative space commands (alias):

\begin{math}
  \mi{nthin}\!
  \mi{end}
\end{math}

\medskip
Negative spaces in subscripts:

\begin{math}
  x_{
    \mi{nthin}\negthinspace
    \mi{nmed}\negmedspace
    \mi{nthick}\negthickspace
    \mi{end}_{
      \mi{nthin}\negthinspace
      \mi{nmed}\negmedspace
      \mi{nthick}\negthickspace
      \mi{end}
    }
  }
\end{math}

\medskip
Varying math font size
\par
\begin[parameter=math.font.size, value=6]{set}
  \begin{math}
    x_{
      \mi{thin}\thinspace
      \mi{med}\medspace
      \mi{thick}\thickspace
      \mi{en}\enspace
      \mi{quad}\quad
      \mi{qquad}\qquad
      \mi{end}_{
        \mi{thin}\thinspace
        \mi{med}\medspace
        \mi{thick}\thickspace
        \mi{en}\enspace
        \mi{quad}\quad
        \mi{qquad}\qquad
        \mi{end}
      }
    }
  \end{math}
\end{set}

\par
\begin[parameter=math.font.size, value=18]{set}
  \begin{math}
    x_{
      \mi{thin}\thinspace
      \mi{med}\medspace
      \mi{thick}\thickspace
      \mi{en}\enspace
      \mi{quad}\quad
      \mi{qquad}\qquad
      \mi{end}_{
        \mi{thin}\thinspace
        \mi{med}\medspace
        \mi{thick}\thickspace
        \mi{en}\enspace
        \mi{quad}\quad
        \mi{qquad}\qquad
        \mi{end}
      }
    }
  \end{math}
\end{set}

\end{document}
