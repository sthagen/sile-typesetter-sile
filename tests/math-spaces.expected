Set paper size 	595.275597	841.8897729
Begin page
Mx 	49.7638
My 	49.6628
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	51 82 86 76 87 76 89 72 w=32.5342	(Positive)
Mx 	84.9855
T	86 83 68 70 72 w=22.7393	(space)
Mx 	110.4123
T	70 82 80 80 68 81 71 86 w=44.6338	(commands)
Mx 	155.0460
T	29 w=2.2900	(:)
Mx 	49.7638
My 	61.6628
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	85 73 74 79 w=16.6700	(nil)
Mx 	68.1004
Mx 	7.9000
Mx 	4.5400
Mx 	5.0600
T	78 a=7.9000 70 a=4.4700 69 a=5.0600	(nil)
Mx 	87.7915
T	85 73 74 68 76 w=20.6500	(nil)
Mx 	111.3163
T	70 79 w=9.8900	(nil)
Mx 	126.2063
T	82 86 66 69 w=19.9700	(nil)
Mx 	156.1763
T	82 82 86 66 69 w=25.0000	(nil)
Mx 	201.1763
T	70 79 69 w=14.9500	(nil)
Mx 	49.7638
My 	79.6628
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	51 82 86 76 87 76 89 72 w=32.5342	(Positive)
Mx 	84.9815
T	86 83 68 70 72 w=22.7393	(space)
Mx 	110.4043
T	70 82 80 80 68 81 71 86 w=44.6338	(commands)
Mx 	157.7217
T	11 w=3.1689	(()
Mx 	160.8906
T	68 79 76 68 86 72 86 w=26.9434	(aliases)
Mx 	187.8340
T	12 w=3.1689	())
Mx 	191.0029
T	29 w=2.2900	(:)
Mx 	49.7638
My 	91.6628
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	85 73 74 79 w=16.6700	(nil)
Mx 	68.1004
Mx 	7.9000
Mx 	4.5400
Mx 	5.0600
T	78 a=7.9000 70 a=4.4700 69 a=5.0600	(nil)
Mx 	87.7998
T	85 73 74 68 76 w=20.6500	(nil)
Mx 	111.3455
T	70 79 w=9.8900	(nil)
Mx 	126.2355
T	70 79 69 w=14.9500	(nil)
Mx 	49.7638
My 	109.6628
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	51 82 86 76 87 76 89 72 w=32.5342	(Positive)
Mx 	84.9842
T	86 83 68 70 72 86 w=26.6016	(spaces)
Mx 	114.2719
T	76 81 w=8.2275	(in)
Mx 	125.1857
T	86 88 69 86 70 85 76 83 87 86 w=41.7432	(subscripts)
Mx 	166.9289
T	29 w=2.2900	(:)
Mx 	49.7638
My 	121.6628
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	2768 w=5.4100	(nil)
Mx 	54.3638
My 	123.7628
Set font 	Libertinus Math;8;400;Regular;normal;+ssty=1;;LTR
T	85 73 74 79 w=13.3360	(nil)
Mx 	69.0331
Mx 	6.3200
Mx 	3.6320
Mx 	4.0480
T	78 a=6.3200 70 a=3.5760 69 a=4.0480	(nil)
Mx 	84.7796
T	85 73 74 68 76 w=16.5200	(nil)
Mx 	103.5834
T	70 79 w=7.9120	(nil)
Mx 	115.4954
T	82 86 66 69 w=15.9760	(nil)
Mx 	139.4714
T	82 82 86 66 69 w=20.0000	(nil)
Mx 	175.4714
T	70 79 69 w=11.9600	(nil)
Mx 	187.4314
My 	125.4428
Set font 	Libertinus Math;6;400;Regular;normal;+ssty=2;;LTR
T	85 73 74 79 w=10.0020	(nil)
Mx 	198.4334
Mx 	4.7400
Mx 	2.7240
Mx 	3.0360
T	78 a=4.7400 70 a=2.6820 69 a=3.0360	(nil)
Mx 	210.2433
T	85 73 74 68 76 w=12.3900	(nil)
Mx 	224.3462
T	70 79 w=5.9340	(nil)
Mx 	233.2802
T	82 86 66 69 w=11.9820	(nil)
Mx 	251.2622
T	82 82 86 66 69 w=15.0000	(nil)
Mx 	278.2622
T	70 79 69 w=8.9700	(nil)
Mx 	49.7638
My 	141.4152
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	49 72 74 68 87 76 89 72 w=36.2793	(Negative)
Mx 	88.7302
T	86 83 68 70 72 w=22.7393	(space)
Mx 	114.1566
T	70 82 80 80 68 81 71 86 w=44.6338	(commands)
Mx 	158.7903
T	29 w=2.2900	(:)
Mx 	49.7638
My 	153.4152
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	79 85 73 74 79 w=22.0900	(nil)
Mx 	70.1871
Mx 	5.4200
Mx 	7.9000
Mx 	4.5400
Mx 	5.0600
T	79 a=5.4200 78 a=7.9000 70 a=4.4700 69 a=5.0600	(nil)
Mx 	90.8149
T	79 85 73 74 68 76 w=26.0700	(nil)
Mx 	114.1071
T	70 79 69 w=14.9500	(nil)
Mx 	49.7638
My 	171.4152
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	49 72 74 68 87 76 89 72 w=36.2793	(Negative)
Mx 	88.7272
T	86 83 68 70 72 w=22.7393	(space)
Mx 	114.1505
T	70 82 80 80 68 81 71 86 w=44.6338	(commands)
Mx 	161.4684
T	11 w=3.1689	(()
Mx 	164.6373
T	68 79 76 68 86 w=18.4619	(alias)
Mx 	183.0992
T	12 w=3.1689	())
Mx 	186.2682
T	29 w=2.2900	(:)
Mx 	49.7638
My 	183.4152
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	79 85 73 74 79 w=22.0900	(nil)
Mx 	70.1871
T	70 79 69 w=14.9500	(nil)
Mx 	49.7638
My 	201.4152
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	49 72 74 68 87 76 89 72 w=36.2793	(Negative)
Mx 	88.7289
T	86 83 68 70 72 86 w=26.6016	(spaces)
Mx 	118.0162
T	76 81 w=8.2275	(in)
Mx 	128.9296
T	86 88 69 86 70 85 76 83 87 86 w=41.7432	(subscripts)
Mx 	170.6727
T	29 w=2.2900	(:)
Mx 	49.7638
My 	213.4152
Set font 	Libertinus Math;10;400;Regular;normal;;;LTR
T	2768 w=5.4100	(nil)
Mx 	54.3638
My 	215.5152
Set font 	Libertinus Math;8;400;Regular;normal;+ssty=1;;LTR
T	79 85 73 74 79 w=17.6720	(nil)
Mx 	70.7024
Mx 	4.3360
Mx 	6.3200
Mx 	3.6320
Mx 	4.0480
T	79 a=4.3360 78 a=6.3200 70 a=3.5760 69 a=4.0480	(nil)
Mx 	87.2047
T	79 85 73 74 68 76 w=20.8560	(nil)
Mx 	105.8384
T	70 79 69 w=11.9600	(nil)
Mx 	117.7984
My 	217.1952
Set font 	Libertinus Math;6;400;Regular;normal;+ssty=2;;LTR
T	79 85 73 74 79 w=13.2540	(nil)
Mx 	130.0524
Mx 	3.2520
Mx 	4.7400
Mx 	2.7240
Mx 	3.0360
T	79 a=3.2520 78 a=4.7400 70 a=2.6820 69 a=3.0360	(nil)
Mx 	142.4291
T	79 85 73 74 68 76 w=15.6420	(nil)
Mx 	156.4044
T	70 79 69 w=8.9700	(nil)
Mx 	49.7638
My 	231.8416
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	57 68 85 92 76 81 74 w=33.0859	(Varying)
Mx 	85.5386
T	80 68 87 75 w=21.5820	(math)
Mx 	109.8095
T	73 82 81 87 w=17.0996	(font)
Mx 	129.5980
T	86 76 93 72 w=15.4346	(size)
Mx 	49.7638
My 	243.8416
Set font 	Libertinus Math;6;400;Regular;normal;;;LTR
T	2768 w=3.2460	(nil)
Mx 	52.5238
My 	245.1016
Set font 	Libertinus Math;4.8;400;Regular;normal;+ssty=1;;LTR
T	85 73 74 79 w=8.0016	(nil)
Mx 	61.3254
Mx 	3.7920
Mx 	2.1792
Mx 	2.4288
T	78 a=3.7920 70 a=2.1456 69 a=2.4288	(nil)
Mx 	70.7773
T	85 73 74 68 76 w=9.9120	(nil)
Mx 	82.0697
T	70 79 w=4.7472	(nil)
Mx 	90.8169
T	82 86 66 69 w=9.5856	(nil)
Mx 	108.4025
T	82 82 86 66 69 w=12.0000	(nil)
Mx 	136.4025
T	70 79 69 w=7.1760	(nil)
Mx 	143.5785
My 	246.1096
Set font 	Libertinus Math;3.6;400;Regular;normal;+ssty=2;;LTR
T	85 73 74 79 w=6.0012	(nil)
Mx 	150.1797
Mx 	2.8440
Mx 	1.6344
Mx 	1.8216
T	78 a=2.8440 70 a=1.6092 69 a=1.8216	(nil)
Mx 	157.2686
T	85 73 74 68 76 w=7.4340	(nil)
Mx 	165.7380
T	70 79 w=3.5604	(nil)
Mx 	172.2984
T	82 86 66 69 w=7.1892	(nil)
Mx 	185.4876
T	82 82 86 66 69 w=9.0000	(nil)
Mx 	206.4876
T	70 79 69 w=5.3820	(nil)
Mx 	49.7638
My 	255.8540
Set font 	Libertinus Math;18;400;Regular;normal;;;LTR
T	2768 w=9.7380	(nil)
Mx 	58.0438
My 	259.6340
Set font 	Libertinus Math;14.4;400;Regular;normal;+ssty=1;;LTR
T	85 73 74 79 w=24.0048	(nil)
Mx 	84.4486
Mx 	11.3760
Mx 	6.5376
Mx 	7.2864
T	78 a=11.3760 70 a=6.4368 69 a=7.2864	(nil)
Mx 	112.7680
T	85 73 74 68 76 w=29.7360	(nil)
Mx 	146.5545
T	70 79 w=14.2416	(nil)
Mx 	164.7961
T	82 86 66 69 w=28.7568	(nil)
Mx 	201.5529
T	82 82 86 66 69 w=36.0000	(nil)
Mx 	253.5529
T	70 79 69 w=21.5280	(nil)
Mx 	275.0809
My 	262.6580
Set font 	Libertinus Math;10.8;400;Regular;normal;+ssty=2;;LTR
T	85 73 74 79 w=18.0036	(nil)
Mx 	294.8845
Mx 	8.5320
Mx 	4.9032
Mx 	5.4648
T	78 a=8.5320 70 a=4.8276 69 a=5.4648	(nil)
Mx 	316.1240
T	85 73 74 68 76 w=22.3020	(nil)
Mx 	341.4639
T	70 79 w=10.6812	(nil)
Mx 	355.1451
T	82 86 66 69 w=21.5676	(nil)
Mx 	382.7127
T	82 82 86 66 69 w=27.0000	(nil)
Mx 	421.7127
T	70 79 69 w=16.1460	(nil)
Mx 	295.2916
My 	780.6177
Set font 	Gentium Plus;10;400;;normal;;;LTR
T	20 w=4.6924	(1)
End page
Finish
